packages/gatsby/src/utils/page-data.ts

```diff
@@ -178,3 +180,50 @@ export function enqueueFlush(): void {
     flush()
   }
 }
+
+export async function handleStalePageData(): Promise<void> {
+  if (!(await fs.pathExists(`public/page-data`))) {
+    return
+  }
+
+  // public directory might have stale page-data files from previous builds
+  // we get the list of those and compare against expected page-data files
+  // and remove ones that shouldn't be there anymore
+
+  const activity = reporter.activityTimer(`Cleaning up stale page-data`)
+  activity.start()
+
+  const pageDataFilesFromPreviousBuilds = await new Promise<Set<string>>(
```

Comment:
I think you should hook in the error event for the stream, at least, and `reject` for that case.

Reply:
listening to `error` events and rejecting: https://github.com/gatsbyjs/gatsby/pull/26937/commits/424d840a10cb2e916e94548d39132280be575d6a

I don't really know how could I handle this - it does seems fatal if we can't traverse `public/page-data`, so the rejection I added in https://github.com/gatsbyjs/gatsby/pull/26937/commits/424d840a10cb2e916e94548d39132280be575d6a will do that